src/index.js

```diff
@@ -1,17 +1,163 @@
+    this.firefox = this.name === 'firefox';
+    this.safari = this.name === 'safari';
+    this.edge = this.name === 'microsoft edge';
+    this.ie = this.name === 'internet explorer';
+    this.opera = this.name === 'opera';
+    this.node = this.name === 'node';
+    this.instagram = /instagram/i.test(detector.ua);
+    this.pinterest = /pinterest/i.test(detector.ua);
+    this.facebook = /fban|fbav/i.test(detector.ua);
+    this.linkedIn = /linkedin/i.test(detector.ua);
+    this.snapchat = /snapchat/i.test(detector.ua);
+    this.whatsApp = /whatsapp/i.test(detector.ua);
+    this.twitter = /twitter/i.test(detector.ua);
+    this.weChat = /wechat|micromessenger/i.test(detector.ua);
+    this.tikTok = /musical_ly/i.test(detector.ua);
+    this.inApp =
+      this.instagram ||
+      this.pinterest ||
+      this.facebook ||
+      this.linkedIn ||
+      this.snapchat ||
+      this.whatsApp ||
+      this.twitter ||
+      this.weChat ||
+      this.tikTok ||
+      (os.ios && os.majorVersion >= 11 && this.safari && typeof detector.window.navigator.mediaDevices === 'undefined');
+  }
+  get version() {
+    return detector.base.getBrowserVersion() || 0;
+  }
+  get majorVersion() {
+    return parseInt(this.version.replace(/[^0-9.]/g, ''), 10);
+  }
+}
+export const browser = new Browser();
+
+class Device {
+  constructor() {
+    this.platform = detector.isSpoofedIpad
+      ? { type: 'tablet', vendor: 'Apple', model: 'iPad' }
+      : detector.base.getPlatform();
+    this.type = (this.platform.type || '').toLowerCase();
+    this.model = (this.platform.model || '').toLowerCase();
+    this.phone = this.type === 'mobile';
+    this.tablet = this.type === 'tablet';
+    this.mobile = this.phone || this.tablet;
+    this.desktop = !this.mobile;
+    this.iphone = this.model === 'iphone';
+    this.ipad = this.model === 'ipad';
+    this.ipod = this.model === 'ipod';
+    this.pixelRatio = detector.window.devicePixelRatio;
+    this.node = this.type === 'node';
+    this.browser = !this.node;
+  }
+  get orientation() {
+    if (typeof detector.window.screen === 'object') {
+      const orientationType =
+        detector.window.screen.msOrientation ||
+        (detector.window.screen.orientation || detector.window.screen.mozOrientation || {}).type;
+      if (typeof orientationType === 'string') {
+        return orientationType.split('-', 1)[0];
+      }
+    }
+    if (typeof detector.window.matchMedia === 'function') {
+      return detector.window.matchMedia('(orientation: portrait)').matches === true ? 'PORTRAIT' : 'LANDSCAPE';
+    }
+    const w = Math.max(document.documentElement.clientWidth, detector.window.innerWidth || 0);
+    const h = Math.max(document.documentElement.clientHeight, detector.window.innerHeight || 0);
+    return w < h ? 'PORTRAIT' : 'LANDSCAPE';
+  }
+  get portrait() {
+    return this.orientation === 'PORTRAIT';
+  }
+  get landscape() {
+    return this.orientation === 'LANDSCAPE';
+  }
```

Comment:
@edgardz  we are currently relying on the `useOrientation` hook from `react-use` library. Do you have any other good potential use cases if we have this in the `detect` library?